js/clinicalTimeline.js

```diff
@@ -51,6 +55,33 @@ window.clinicalTimeline = (function(){
       beginning = "0";
     }
 
+    var zoomLevel = getZoomLevel(beginning, ending, width * zoomFactor);
+    
+    switch(zoomLevel){
+      case "years" : 
+        var beginningTick = roundDown(beginTime, zoomLevel) * beginTime.daysPerYear;
+        var endingTick = roundUp(endTime, zoomLevel) * endTime.daysPerYear;
+        break;
+      case "months" :
+        var beginningTick = roundDown(beginTime, zoomLevel) * beginTime.daysPerMonth - 5;
+        var endingTick = roundUp(endTime, zoomLevel) * endTime.daysPerMonth;
+        break;
+      case "10days" :
+        var beginningTick = roundDown(beginTime, zoomLevel);
+        var endingTick = roundUp(endTime, zoomLevel);
+        break;
+      case "3days" :
+        var beginningTick = roundDown(beginTime, zoomLevel);
+        var endingTick = roundUp(endTime, zoomLevel);
+        break;
+      case "days" :
+        var beginningTick = roundDown(beginTime, zoomLevel);
+        var endingTick = roundUp(endTime, zoomLevel);
+        break; 
+      default :
+        console.error("Undefined zoomLevel");
+    }
+    
```

Comment:
It would seem you are trying to fix the same problem twice. Determining the `beginningTick` and `endingTick` is now done here in `timeline()` and in `getTickValues`. Either one would determine them before calling `getTickValues` and assume inside `getTickValues` you have the correct ones. Or you call `getTickValues` and use the first and last item from that array as you did before. I think the latter makes the most sense since it would seem that `getTickValues` already determines the correct first and last tick. Another thing to note is that there seem to be a bunch of different variable names for the same thing now. We have both `beginning` and `beginningTick` for the first tick. We also use `beginTime`, `minTime`, `endTime`, `maxTime`. That's a little confusing to me.